### ISSUE TITLE
Support provided.al2023 runtime

### ISSUE THREAD

**Issue by davidjb** [aws-cloudformation/cfn-lint#2945]
### CloudFormation Lint Version

cfn-lint 0.83.1 (but built from latest main @ https://github.com/aws-cloudformation/cfn-lint/commit/6f5c89f6741aa4c03300b2c666e1382f5502fcd4)

### What operating system are you using?

Ubuntu

### Describe the bug

`provided.al2023` (https://aws.amazon.com/blogs/compute/introducing-the-amazon-linux-2023-runtime-for-aws-lambda/) is not currently supported as a valid runtime by `cfn-lint`:

```
E3030 You must specify a valid value for Runtime (provided.al2023). Valid values are ["dotnet6", "dotnetcore1.0", "dotnetcore2.0", "dotnetcore2.1", "dotnetcore3.1", "go1.x", "java11", "java17", "java8", "java8.al2", "nodejs", "nodejs10.x", "nodejs12.x", "nodejs14.x", "nodejs16.x", "nodejs18.x", "nodejs20.x", "nodejs4.3", "nodejs4.3-edge", "nodejs6.10", "nodejs8.10", "provided", "provided.al2", "python2.7", "python3.10", "python3.11", "python3.6", "python3.7", "python3.8", "python3.9", "ruby2.5", "ruby2.7", "ruby3.2"]
/
```

### Expected behavior

Template to validate correctly

### Reproduction template

```yaml
AWSTemplateFormatVersion: "2010-09-09"
Transform: AWS::Serverless-2016-10-31
Resources:
  MyFunction:
    Type: AWS::Serverless::Function
    Metadata:
      BuildMethod: go1.x
    Properties:
      CodeUri: myfunction/
      Handler: bootstrap
      Runtime: provided.al2023
```

**Comment by kddejong** [comment]
We get the values from https://github.com/boto/botocore/blob/50861b96f9d51632d6ddd8bafac8d50f4e8b027a/botocore/data/lambda/2015-03-31/service-2.json#L5200 and is included in pr #2943

**Comment by kddejong** [comment]
Merged and release with v0.83.2
